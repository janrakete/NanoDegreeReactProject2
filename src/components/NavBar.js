import React from "react";
import { connect } from "react-redux";
import { NavLink, useNavigate } from "react-router-dom";

import { UserLogoutHandle } from "../actions/UserLoginLogout";

function NavBar({ dispatch, UserLoginLogout }) {

  const navigate = useNavigate();

	const LogoutHandleSubmit = () => {
		dispatch(UserLogoutHandle());
		navigate("/");
	};

  return (
	<div>
		<div className="navbar">
			<div className="navbar-item navbar-brand" id="navbar">The Poll-utor</div>
			<div className="navbar-item">
				<NavLink to="/" exact="true">
					Dashboard
				</NavLink>
			</div>
			<div className="navbar-item">
				<NavLink to="/leaderboard" exact="true">
					Leaderboard
				</NavLink>
			</div>
			<div className="navbar-item navbar-highlight">
				<NavLink to="/add" exact="true">
					New poll
				</NavLink>
				</div>
		</div>
		<div className="navbar-user">
			<div className="navbar-item">Logged in as: <img src={UserLoginLogout.avatarURL} width="25" valign="middle" alt="img" /> {UserLoginLogout.id}</div>
			<div className="navbar-item navbar-highlight"><a href="#" onClick={LogoutHandleSubmit}>Logout</a></div>
		</div>  
	</div>    
    )
  }

function mapStateToProps({ UserLoginLogout }) {
  return {
    UserLoginLogout,
  };
}

export default connect(mapStateToProps)(NavBar);