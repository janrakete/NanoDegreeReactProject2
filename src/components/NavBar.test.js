import { fireEvent, render, screen, waitFor, waitForElementToBeRemoved, findByText } from "@testing-library/react";
import { MemoryRouter as Router } from "react-router-dom";
import { Provider } from "react-redux";

import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
import NavBar from "./NavBar";

const mockStore = configureStore([thunk]);

describe("NavBar", () => {
  const USER_NAME = "Mike Tsamis";
  const USER_ID = "mtsamis";
  const QUESTION_ID = "6ni6ok3ym7mf1p33lnez";
  const USER_PASSWORD = "xyz123";
  
  let store, component;

  beforeEach(() => {
    store = mockStore({
      UserLoginLogout: {
          id: USER_ID,
          name: USER_NAME,
          avatarURL: "https://www.gravatar.com/avatar/00000000000000000000000000000000?d=retro&f=y",
          password : USER_PASSWORD,
      },
    });

    component = render(
      <Provider store={store}>
        <Router>
          <NavBar />
        </Router>
      </Provider>
    );
  });

  // Unit testing Case 5
  it("should match the navbar snapshot with logged in user mtsamis", () => {
    expect(component).toMatchSnapshot();
  });
});


