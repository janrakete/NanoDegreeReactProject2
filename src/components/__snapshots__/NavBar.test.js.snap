// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar should match the navbar snapshot with logged in user mtsamis 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="navbar"
        >
          <div
            class="navbar-item navbar-brand"
            id="navbar"
          >
            The Poll-utor
          </div>
          <div
            class="navbar-item"
          >
            <a
              aria-current="page"
              class="active"
              exact="true"
              href="/"
            >
              Dashboard
            </a>
          </div>
          <div
            class="navbar-item"
          >
            <a
              class=""
              exact="true"
              href="/leaderboard"
            >
              Leaderboard
            </a>
          </div>
          <div
            class="navbar-item navbar-highlight"
          >
            <a
              class=""
              exact="true"
              href="/add"
            >
              New poll
            </a>
          </div>
        </div>
        <div
          class="navbar-user"
        >
          <div
            class="navbar-item"
          >
            Logged in as: 
            <img
              alt="img"
              src="https://www.gravatar.com/avatar/00000000000000000000000000000000?d=retro&f=y"
              valign="middle"
              width="25"
            />
             
            mtsamis
          </div>
          <div
            class="navbar-item navbar-highlight"
          >
            <a
              href="#"
            >
              Logout
            </a>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="navbar"
      >
        <div
          class="navbar-item navbar-brand"
          id="navbar"
        >
          The Poll-utor
        </div>
        <div
          class="navbar-item"
        >
          <a
            aria-current="page"
            class="active"
            exact="true"
            href="/"
          >
            Dashboard
          </a>
        </div>
        <div
          class="navbar-item"
        >
          <a
            class=""
            exact="true"
            href="/leaderboard"
          >
            Leaderboard
          </a>
        </div>
        <div
          class="navbar-item navbar-highlight"
        >
          <a
            class=""
            exact="true"
            href="/add"
          >
            New poll
          </a>
        </div>
      </div>
      <div
        class="navbar-user"
      >
        <div
          class="navbar-item"
        >
          Logged in as: 
          <img
            alt="img"
            src="https://www.gravatar.com/avatar/00000000000000000000000000000000?d=retro&f=y"
            valign="middle"
            width="25"
          />
           
          mtsamis
        </div>
        <div
          class="navbar-item navbar-highlight"
        >
          <a
            href="#"
          >
            Logout
          </a>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
